src/phoebus_guibuilder/guibuilder.py

```diff
@@ -1,171 +1,129 @@
-                [\d]*   # match zero or more digits
-                [^:]?   # match zero or one non-colon character
-            )
-            (?::([a-zA-Z0-9:]*))? # match zero or one colon followed by zero or more
-                                # alphanumeric characters or colons (capture group 2)
-            (?:\.([a-zA-Z0-9]+))? # match zero or one dot followed by one or more
-                                # alphanumeric characters (capture group 3)
-            $           # end of string
-        """,
-            re.VERBOSE,
-        )
-
-        match = re.match(pattern, self.prefix)
-        if match:
-            self.P: str = match.group(1)
-            self.R: str = match.group(2)
-            # TODO: Is this needed?
-            self.attribute: str | None = match.group(3)
-        else:
-            raise AttributeError(f"No valid PV prefix found for {self.name}.")
-
-
-def main(filename: str):
-    components: list[Component] = []  # TODO Manage global lists better
-
-    with open(filename) as f:
-        conf = yaml.safe_load(f)
-
-        bl: dict[str, str] = conf["beamline"]
-        comps: dict[str, dict[str, str]] = conf["components"]
-
-        beamline = Beamline(**bl)
-
-        for key, comp in comps.items():
-            components.append(Component(key, **comp))
-
-    print("BEAMLINE:")
-    pp.pprint(beamline)
-
-    print("")
-    print("COMPONENTS")
-    pp.pprint(components)
-
-
-#####################################################
-# TODO Functionality should be in phoebusguibuilder class
-# class Phoebusguibuilder(beamline: Beamline, components: list[Component]):
-
-
-def find_services_folders(beamline: Beamline, components: list[Component]):
-    services_directory = (
-        beamline.dom + "-services/services"
-    )  # TODO: rm hardcoding, map to services.
-    path = f"/dls/science/users/uns32131/{services_directory}"
-    files = os.listdir(path)
-
-    # Attempting to match the prefix to the files in the services directory
-    pattern = "^(.*)-(.*)-(.*)"
-
-    for component in components:
-        domain: re.Match[str] | None = re.match(pattern, component.P)
-        assert domain is not None, "Empty Prefix Field"
-
-        for file in files:
-            match = re.match(pattern, file)
-            if match:
-                if match.group(1) == domain.group(1).lower():
-                    if os.path.exists(f"{path}/{file}/config/ioc.yaml"):
-                        valid_entities = extract_valid_entities(
-                            ioc_yaml=f"{path}/{file}/config/ioc.yaml",
-                            component=component,
+from phoebus_guibuilder.datatypes import Beamline, Component, Entry
+
+
+class Guibuilder:
+    """
+    This class provides the functionality to process the required
+    create_gui.yaml file into screens mapped from ioc.yaml and
+    gui_map.yaml files.
+
+    """
+
+    def __init__(self, create_gui: str):
```

Comment:
It might be better to name the file arg something like "create_gui_file" or "create_gui_yaml" just to clarify it is the file.